proto/proto/store/rpc.proto

```diff
@@ -164,19 +164,22 @@ message AccountProofResponse {
         // Account storage data
         AccountStorageDetails storage_details = 2;
 
-        // Account code; empty if code commitments matched or none was requested
+        // Account code; empty if code commitments matched or none was requested.
         optional bytes code = 3;
 
         // Account asset vault data; empty if vault commitments matched or the requester
         // omitted it in the request.
         optional AccountVaultDetails vault_details = 4;
     }
 
-    // Account ID, current state commitment, and SMT path
-    account.AccountWitness witness = 1;
+    // The block number at which the account witness and details were observed.
+    blockchain.BlockNumber block_num = 1;
```

Comment:
I think we should make `block_num` here optional. If `block_num` is not provided, we'd just return the data for the latest block. Also, as I mentioned in one of my previous comments, for now, if `block_num` is provided, we'll return an error and then, in v0.13 we'll enable the full functionality.